Framework/API/src/FunctionDomain1D.cpp

```diff
@@ -127,5 +127,37 @@ FunctionDomain1DSpectrum::FunctionDomain1DSpectrum(
     std::vector<double>::const_iterator to)
     : FunctionDomain1DVector(from, to), m_workspaceIndex(wi) {}
 
+/// Constructor.
+/// @param bins :: A vector with bin boundaries.
+FunctionDomain1DHistogram::FunctionDomain1DHistogram(
+    const std::vector<double> &bins)
```

Comment:
This looks identical to the other constructor, I think you can just delegate to the other constructor:

```cpp
  : FunctionDomain1DHistogram(bins.begin(), bins.end())) {}
```